### PR TITLE
TYP: Type annotations in pandas/io/formats/printing.py

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/io/formats/printing.py [hunk 1]
```diff
@@ -182,13 +182,12 @@ def pprint_thing(
         replacements
     default_escapes : bool, default False
         Whether the input escape characters replaces or adds to the defaults
-    max_seq_items : False, int, default None
+    max_seq_items : int, default None
```
Comment (jbrockmendel) [inline pandas/io/formats/printing.py]: "int or None, default None"?

File: pandas/io/formats/printing.py [hunk 2]
```diff
@@ -346,15 +344,17 @@ def format_object_summary(
...
+    ) -> Tuple:
 
         if adj.len(line.rstrip()) + adj.len(value.rstrip()) >= display_width:
             s += line.rstrip()
             line = next_line_prefix
         line += value
         return s, line
 
-    def best_len(values):
+    def best_len(values) -> int:
```
Comment (jbrockmendel) [inline pandas/io/formats/printing.py]: can the values arg be typed?

File: pandas/io/formats/printing.py [hunk 3]
```diff
@@ -346,15 +344,17 @@ def format_object_summary(
     # adj can optionally handle unicode eastern asian width
     adj = _get_adjustment()
 
-    def _extend_line(s, line, value, display_width, next_line_prefix):
+    def _extend_line(
+        s, line: str, value: str, display_width: int, next_line_prefix: str
+    ) -> Tuple:
```
Comment (jbrockmendel) [inline pandas/io/formats/printing.py]: can we say anything about what the Tuple contains?

File: pandas/io/formats/printing.py [hunk 4]
```diff
@@ -182,13 +182,12 @@ def pprint_thing(
         replacements
     default_escapes : bool, default False
         Whether the input escape characters replaces or adds to the defaults
-    max_seq_items : False, int, default None
+    max_seq_items : int, default None
         Pass thru to other pretty printers to limit sequence printing
```
Comment (jbrockmendel) [inline pandas/io/formats/printing.py]: thru -> through

**Comment by jbrockmendel** [comment]
Couple of comments, generally looks good.  I'm not sure, but it _may_ be the case that since py2 was dropped some usages of pprint_thing throughout the code base are now unnecessary

File: pandas/io/formats/printing.py (hunk 2 above)
Comment (MomIsBestFriend) [reply pandas/io/formats/printing.py]: mypy suggest typing it as ```List[Any]```

File: pandas/io/formats/printing.py [hunk 5]
```diff
@@ -346,15 +344,17 @@ def format_object_summary(
...
+    ) -> Tuple[str, str]:
 
         if adj.len(line.rstrip()) + adj.len(value.rstrip()) >= display_width:
             s += line.rstrip()
             line = next_line_prefix
         line += value
         return s, line
 
-    def best_len(values):
+    def best_len(values: List[Any]) -> int:
```
Comment (WillAyd) [inline pandas/io/formats/printing.py]: `List[Any]` appeases a type checker but is pretty loose. Is there a way to refine the subtype further?

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/io/formats/printing.py (hunk 5 above)
Comment (MomIsBestFriend) [reply pandas/io/formats/printing.py]: I also don't like this solution, but "mypy " complains about it with every other annotations, that I tried.

Do you have any suggestions?

Comment (WillAyd) [reply pandas/io/formats/printing.py]: `List[str]` maybe

Comment (MomIsBestFriend) [reply pandas/io/formats/printing.py]: > `List[str]` maybe

It worked!

I wonder why it didn't worked for me yesterday, maybe I was annotating other things and reverted without checking it... anyway, thank you :)

**Comment by jreback** [comment]
thanks @MomIsBestFriend

**Pull request merged**